lib/ably/models/message_encoders/vcdiff.rb

```diff
@@ -0,0 +1,35 @@
+require 'ably/models/message_encoders/base'
+
+module Ably::Models::MessageEncoders
+  # Delta Vcdiff Encoder and Decoder
+  # Uses encoding identifier 'vcdiff' and encodes all JSON objects in Vcdiff format, and sets the encoding when decoding
+  #
+  class Vcdiff < Base
+    ENCODING_ID = 'vcdiff'
+
+    def encode(message, channel_options)
+
+    end
+
+    def decode(message, channel_options)
+      return nil unless current_encoding_part(message).to_s.match(/^#{ENCODING_ID}$/i)
+
+      if !channel_options[:plugins] || !(vcdiff = channel_options[:plugins][:vcdiff]) # (PC3)
+        raise Ably::Exceptions::VcdiffError.new('Missing vcdiff decoder (https://github.com/ably-forks/vcdiff-decoder-ruby)', 400, 40019)
```

Comment:
Thanks @owenpearson for the comment. 

Here is an example Vcdiff plugin implementation:

```ruby
require 'VCDiff' # gem romanbsd-vcdiff

class VcdiffPlugin
  def self.decode(data, dictionary)
    @decoder ||= VCDiff::Decoder.new
    @decoder.decode(dictionary, data)
  end
end
```


and example snippet, how to use it:

```ruby
subscriber = Thread.new do
  EventMachine.run do
    ably = Ably::Realtime.new(key: ENV['ABLY_API_KEY'], log_level: :debug, plugins: { vcdiff: VcdiffPlugin }) # (1)
    channel = ably.channels.get 'channel-123', params: { delta: 'vcdiff' } # (2)
    channel.subscribe do |data|
      puts data
    end
  end
end

publisher = Thread.new do
  EventMachine.run do
    ably = Ably::Realtime.new(key: ENV['ABLY_API_KEY'], log_level: :debug)
    channel = ably.channels.get 'channel-123', params: { delta: 'vcdiff' } # (2)
    channel.publish 'event1', 'this is example message'
  end
end

subscriber.join
publisher.join
```


It requires external gems (i.e: romanbsd-vcdiff or vcdiff.rb). 

IMO, it is not a good idea to add it to the lib because it adds additional dependencies but also additional work in the future (to maintain: tests, changing ruby version and testing again, etc.). Developers can add it if they need and implement a few lines of code (like in `VcdiffPlugin` class).

It would be good to use vcdiff implementation written in C ext. The fastest implementation of vcdiff decoder is in C ext, it requires native extensions to be installed when installing the gem. I think developers could save some memory and disk storage on IoT. So, I think it would be better to not include it as a dependency.

I can removed ably-forks link, or build own ably vcdiff decoder and then it would be as easy as:

```ruby
require 'ably-ruby-vcdiff'

ably = Ably::Realtime.new(key: ENV['ABLY_API_KEY'], log_level: :debug, plugins: { vcdiff: Ably::Vcdiff }) # (1)

```